util/pkg/hashing/hash.go

```diff
@@ -70,6 +70,19 @@ func (ha HashAlgorithm) NewHasher() hash.Hash {
 	return nil
 }
 
+func ForString(s string) (HashAlgorithm, error) {
```

Comment:
Nit: A comment describing what this function does would be helpful, or renaming it to be more descriptive e.g. `GuessHashAlgorithmFromString`

Reply:
It's a bit roundabout, but I suppose we have `shaHash` lying around.